test/serde/msgpack/test_msgpack_serde_full.py

```diff
@@ -0,0 +1,144 @@
+samples[numpy.int32] = partial(make_numpy_number, numpy.int32)
+samples[numpy.int64] = partial(make_numpy_number, numpy.int64)
+
+# PyTorch
+samples[torch.device] = make_torch_device
+samples[torch.jit.ScriptModule] = make_torch_scriptmodule
+samples[torch._C.Function] = make_torch_cfunction
+samples[torch.jit.TopLevelTracedModule] = make_torch_topleveltracedmodule
+samples[torch.nn.Parameter] = make_torch_parameter
+samples[torch.Tensor] = make_torch_tensor
+samples[torch.Size] = make_torch_size
+
+# PySyft
+samples[
+    syft.frameworks.torch.tensors.interpreters.additive_shared.AdditiveSharingTensor
+] = make_additivesharingtensor
+samples[
+    syft.frameworks.torch.tensors.interpreters.precision.FixedPrecisionTensor
+] = make_fixedprecisiontensor
+samples[
+    syft.frameworks.torch.tensors.interpreters.crt_precision.CRTPrecisionTensor
+] = make_crtprecisiontensor
+samples[syft.frameworks.torch.tensors.decorators.logging.LoggingTensor] = make_loggingtensor
+samples[syft.generic.pointers.multi_pointer.MultiPointerTensor] = make_multipointertensor
+samples[syft.messaging.plan.plan.Plan] = make_plan
+samples[syft.messaging.plan.state.State] = make_state
+samples[syft.messaging.plan.procedure.Procedure] = make_procedure
+samples[syft.messaging.protocol.Protocol] = make_protocol
+samples[syft.generic.pointers.pointer_tensor.PointerTensor] = make_pointertensor
+samples[syft.generic.pointers.pointer_plan.PointerPlan] = make_pointerplan
+samples[syft.generic.pointers.pointer_protocol.PointerProtocol] = make_pointerprotocol
+samples[syft.generic.pointers.object_wrapper.ObjectWrapper] = make_objectwrapper
+samples[syft.generic.pointers.object_pointer.ObjectPointer] = make_objectpointer
+samples[syft.federated.train_config.TrainConfig] = make_trainconfig
+samples[syft.workers.base.BaseWorker] = make_baseworker
+samples[syft.frameworks.torch.tensors.interpreters.autograd.AutogradTensor] = make_autogradtensor
+samples[syft.frameworks.torch.tensors.interpreters.private.PrivateTensor] = make_privatetensor
+samples[syft.frameworks.torch.tensors.interpreters.promise.PromiseTensor] = make_promisetensor
+
+samples[syft.messaging.message.Message] = make_message
+samples[syft.messaging.message.Operation] = make_operation
+samples[syft.messaging.message.ObjectMessage] = make_objectmessage
+samples[syft.messaging.message.ObjectRequestMessage] = make_objectrequestmessage
+samples[syft.messaging.message.IsNoneMessage] = make_isnonemessage
+samples[syft.messaging.message.GetShapeMessage] = make_getshapemessage
+samples[syft.messaging.message.ForceObjectDeleteMessage] = make_forceobjectdeletemessage
+samples[syft.messaging.message.SearchMessage] = make_searchmessage
+samples[syft.messaging.message.PlanCommandMessage] = make_plancommandmessage
+
+samples[syft.exceptions.GetNotPermittedError] = make_getnotpermittederror
+samples[syft.exceptions.ResponseSignatureError] = make_responsesignatureerror
+
+# Dynamically added to msgpack.serde.simplifiers by some other test
```

Comment:
This one can be removed now (was exact same problem you had with Plan, but with BaseWorker, I just added it in the samples list to avoid test failure)

Reply:
Workers have `force_full_simplification`, so we must serialize them intentionally at some point, right? I don't really understand when/why that happens though.